galsim/wfirst/wfirst_detectors.py

```diff
@@ -82,7 +82,28 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
     """
     img.applyIPC(galsim.wfirst.ipc_kernel, edge_treatment=edge_treatment, fill_value=fill_value)
 
-def allDetectorEffects(img, rng=None, exptime=None):
+def applyPersistence(img, prev_exposures):
+    """
+    Applies the persistence effect to the Image instance by adding a small fraction of the previous
+    exposures (up to 8) supplied as the 'prev_exposures' argument.
+
+    For more information about persistence, see the docstring for galsim.Image.applyPersistence.
+    Unlike that routine, this one does not need the coefficients to be specified. However, the list
+    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
+    ignored.
+
+    @param img               The Image to be transformed.
+    @param prev_exposures    List of up to 8 Image instances in the order of exposures, with the
+                             recent exposure being the first element.
+    """
+    if hasattr(prev_exposures,'__iter__'):
+        n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
+        img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
+    else: # only the previous exposure provided
+        img.applyPersistence(prev_exposures,galsim.wfirst.persistence_coefficients[0])
```

Comment:
Done.

On Mon, Apr 25, 2016 at 1:04 PM, Mike Jarvis <notifications@github.com>
wrote:

> In galsim/wfirst/wfirst_detectors.py
> <https://github.com/GalSim-developers/GalSim/pull/710#discussion_r60949369>
> :
>
> > +    exposures (up to 8) supplied as the 'prev_exposures' argument.
> > +
> > +    For more information about persistence, see the docstring for galsim.Image.applyPersistence.
> > +    Unlike that routine, this one does not need the coefficients to be specified. However, the list
> > +    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
> > +    ignored.
> > +
> > +    @param img               The Image to be transformed.
> > +    @param prev_exposures    List of up to 8 Image instances in the order of exposures, with the
> > +                             recent exposure being the first element.
> > +    """
> > +    if hasattr(prev_exposures,'__iter__'):
> > +        n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
> > +        img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
> > +    else: # only the previous exposure provided
> > +        img.applyPersistence(prev_exposures,galsim.wfirst.persistence_coefficients[0])
>
> The doc string says prev_exposures should be a list, so this is
> undocumented behavior. I think rather than document it ("a list of up to 8
> Images or a single Image..."), I think I'd rather just enforce the list API
> both here and in the generic detectors.applyPersistence function. It's not
> so onerous for the user to supply a list of one element, and it would clean
> up the code a bit to not check for both options.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/710/files/0a43752189ac57458edc609eeacd9956f712346f#r60949369>
>



-- 
Regards

Arun Kannawadi Jayaraman
PhD candidate, Dept. of Physics,
Carnegie Mellon University,
Pittsburgh, PA 15213